ReaderApp.js

```diff
@@ -1620,11 +1620,28 @@ class ReaderApp extends React.PureComponent {
     });
   };
   onQueryChange = (type, query, resetQuery, fromBackButton, getFilters) => {
+
+    const updateAggregations = (data) => {
```

Comment:
I think this function should be in the outer scope because it's so large. It's hard to know what dependencies it has on the variables in `onQueryChange` which make it difficult to change.